### PR TITLE
Remove call to mem::uninitialized/zeroed and other improvements

### PR DESCRIPTION
**Pull request by Johannesd3**
This PR is based on @John2143's PR #1, see there for more details. I rebased it on the `v0.1.0` tag which was not in master. I got rid of a dangerous use of `mem::transmute` by using `MaybeUninit`. Additionally, I fixed some deprecations and eventually ran rustfmt.

### CODE REVIEW

File: src/tremor_sys.rs [hunk 1]
```diff
@@ -7,8 +7,8 @@ extern crate libc;
 use libc::{c_int, c_long, c_char, c_void};
 
 pub use ogg_sys::ogg_int64_t;
-pub type ogg_int32_t = libc::int32_t;
```
Comment (hselasky) [inline src/tremor_sys.rs:10]: This is Linuxism. Please keep int32_t .

File: Cargo.toml [hunk 2]
```diff
@@ -7,8 +7,8 @@ description = "Rust bindings to tremor"
 license = "MIT"
 
 [dependencies]
-libc = "*"
```
Comment (hselasky) [inline Cargo.toml:7]: This is also Linuxism. Please revert.

File: Cargo.toml [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+authors = ["Librespot Org"]
+build = "build.rs"
 
 [dependencies]
 libc = "*"
-tremor-sys = { path = "tremor-sys" }
+ogg-sys = "*"
+
+[build-dependencies]
+cc = "1.0.47"
```
Comment (hselasky) [inline Cargo.toml]: Can you explain these versions? What do they mean? Is this cross platform.

File: src/tremor_sys.rs (hunk 1 above)
Comment (Johannesd3) [reply src/tremor_sys.rs:10]: It's simply deprecated: http://rust-lang.github.io/libc/x86_64-unknown-hermit/libc/type.int32_t.html

Comment (hselasky) [reply src/tremor_sys.rs:10]: I see, not a C-language type. Then it is probably fine. Sorry for the noise.

File: Cargo.toml (hunk 3 above)
Comment (Johannesd3) [reply Cargo.toml]: I did not touch the Cargo.toml, it's all from the rebase I made. The current version on crates.io has probably the same Cargo.toml.

Comment (hselasky) [reply Cargo.toml]: I see. Sorry for the noise. Your changes look OK to me then.

**Pull request merged**